src/appleseed/renderer/modeling/light/light.h

```diff
@@ -137,12 +137,15 @@ class APPLESEED_DLLSYMBOL Light
         const foundation::Vector3d&     target,                     // world space target point
         const foundation::Vector3d&     position) const = 0;        // world space emission position
 
+    bool get_cast_shadows() const;
+
   protected:
     int m_flags;
 
   private:
     struct Impl;
     Impl* impl;
+    bool m_cast_shadows;
```

Comment:
The idea of the [PImpl idiom](https://en.cppreference.com/w/cpp/language/pimpl) is to hide implementation details in the .cpp file, so in theory `m_cast_shadows` should go in the PImpl, not here.

However, for performance reasons, we make some exceptions to this rule, as you can see with `m_flags` just above. I suggest moving `m_cast_shadows` next to it.